censustree/iden3tree/tree.go

```diff
@@ -207,23 +217,27 @@ func (t *Tree) Root() []byte {
 	return t.Tree.RootKey().Bytes()
 }
 
-// Dump returns the whole merkle tree serialized in a format that can be used on Import
-func (t *Tree) Dump(root []byte) (keys [][]byte, err error) {
+// Dump returns the whole merkle tree serialized in a format that can be used on Import.
+// TO-DO dump also Values (currently just keys are dumped).
+func (t *Tree) Dump(root []byte) (keys []byte, err error) {
 	rootHash := new(merkletree.Hash)
 	copy(rootHash[:], root)
 	t.updateAccessTime()
 	claims, err := t.Tree.DumpClaims(rootHash)
 	if err != nil {
 		return nil, err
 	}
+	dump := exportData{}
 	for _, c := range claims {
 		cb, err := hex.DecodeString(c)
 		if err != nil {
 			return nil, err
 		}
-		keys = append(keys, cb)
+		ee := exportElement{Key: make([]byte, len(cb)), Value: []byte{}}
+		copy(ee.Key, cb[:])
```

Comment:
I think this copy does not need to happen. cb is just decoded from hex, so it's safe to just use directly.

Reply:
Thought the same but I did not want to risk :sweat_smile:  But since we both got the same feeling, it makes sense to remove the copy. Thanks.

Reply:
Also, you can drop the []byte{} for the empty value, I think.

Reply:
Mmm, I feel like `[]byte{}` is safer than nil. Won't you think so?

Reply:
Not really. The only exception is if the format/API treats both differently and you want the non-nil version, like encoding/json with null vs [].

As far as I can tell, go-bare encodes lists as length + elements, so both should encode the same. In general, I think you should try to do the easiest thing first, which is just using the zero value (nil).

Reply:
ACK